pkg/render/egressgateway/egressgateway.go

```diff
@@ -107,30 +108,23 @@ func (c *component) Objects() ([]client.Object, []client.Object) {
 	objectsToCreate := append(
 		secret.ToRuntimeObjects(c.egwPullSecrets()...),
 		c.egwServiceAccount(),
+		c.egwRole(),
```

Comment:
Again, I think we should only render if necessary.

Reply:
Fixed.